### PR TITLE
Iteration 3 transaction, transaction spec and corresponding test.

### PR DESCRIPTION
**Pull request by Elle-M**
Finalize integrating business analysis.

### CODE REVIEW

File: spec/transaction_repository_spec.rb [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+  describe '#find_by_id' do
+    it 'returns an instance of transaction with a matching id' do
+      transaction_repository.add_to_repo(t_1)
+      transaction_repository.add_to_repo(t_2)
+
+      expect(transaction_repository.find_by_id(6)).to eq(t_1)
+      expect(transaction_repository.find_by_id(1)).to eq(t_2)
+      expect(transaction_repository.find_by_id(9)).to be_nil
+    end
```
Comment (KelsiePorter) [inline spec/transaction_repository_spec.rb:44]: great test coverage here

File: spec/transaction_repository_spec.rb [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  describe '#find_all_by_result' do
+    it 'returns empty array or all with matching result' do
+      transaction_repository.add_to_repo(t_1)
+      transaction_repository.add_to_repo(t_2)
+
+      expect(transaction_repository.find_all_by_result('success')).to eq([t_1])
+      expect(transaction_repository.find_all_by_result('failed')).to eq([t_2])
+      expect(transaction_repository.find_all_by_result('doesntexist')).to eq([])
+    end
+  end
```
Comment (KelsiePorter) [inline spec/transaction_repository_spec.rb:78]: great test coverage for all of these, too!

**Review by KelsiePorter — APPROVED** [review]

File: spec/transaction_repository_spec.rb [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+      transaction_repository.add_to_repo(t_1)
+      transaction_repository.add_to_repo(t_2)
+
+      expect(transaction_repository.all[0].credit_card_number).to eq('4242424242424242')
+      expect(transaction_repository.all[0].credit_card_expiration_date).to eq('0220')
+      expect(transaction_repository.all[0].result).to eq('success')
+
+      transaction_repository.update( 0, {credit_card_number: "4242424242421111"})
+      transaction_repository.update( 0, {credit_card_expiration_date: '0230'})
+      transaction_repository.update( 0, {result: 'success'})
```
Comment (KelsiePorter) [inline spec/transaction_repository_spec.rb:121]: These updates could probably be done all at the same time.

File: spec/transaction_repository_spec.rb [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    end
```
Comment (KelsiePorter) [inline spec/transaction_repository_spec.rb:122]: We should probably have one last set of expectations testing that the updates actually updated the attributes.

**Review by KelsiePorter — COMMENTED** [review]
I left a comment regarding test coverage for #update in transaction_repository_spec.rb

File: spec/transaction_repository_spec.rb (hunk 1 above)
Comment (Elle-M) [reply spec/transaction_repository_spec.rb:44]: Thank you!

File: spec/transaction_repository_spec.rb (hunk 4 above)
Comment (Elle-M) [reply spec/transaction_repository_spec.rb:122]: Agreed. Good catch!

File: spec/transaction_repository_spec.rb (hunk 3 above)
Comment (Elle-M) [reply spec/transaction_repository_spec.rb:121]: I think they should be individual test for line length and to monitor individually.

**Pull request merged**